### PR TITLE
cpu/cortexm: added and use central yield_from_isr() function

### PR DESCRIPTION
**Pull request by haukepetersen**
a) adds a little convenience
b) one step into using the periph drivers without RIOT

### CODE REVIEW

**Comment by kaspar030** [comment]
Did you see #6164?

**Comment by haukepetersen** [comment]
nope, I didn't. But I guess my PR makes the fix significantly easier -> 1 instead of 73 files... 

Adapted this PR to yield_higher()

**Comment by kaspar030** [comment]
> nope, I didn't. But I guess my PR makes the fix significantly easier -> 1 instead of 73 files...

I was referring to the discussion where to place this (common?) code. This PR would only solve it for cortex-m. I have pretty much this PR in a branch, just solving it for all affected architectures.

Also I have a feeling that the way the includes are ordered this way are weird (thread.h includes arch/thread.h, cpu_conf.h, sched.h; now cortexm/cpu.h includes sched.h, thread.h, ...).

I'd say let's close this PR as duplicate and find a common way in #6171 (possibly re-using this PR's commit).

**Comment by haukepetersen** [comment]
I don't care which way we go, I just need it as soon as possible.

**Comment by haukepetersen** [comment]
ok, new plan: I changed this PR back to the beginning, all it does is simply centralizing the ISR end function. As talked offline with @kaspar030: let's merge this and than use #6164 to fix to `yield_higher` and decide if we want to move that function somewhere else.

File: cpu/cc2538/periph/spi.c [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 #include "cpu.h"
 #include "mutex.h"
+#include "thread.h"
```
Comment (lebrush) [inline cpu/cc2538/periph/spi.c]: Thread is included twice now in this file (see L31)

Comment (haukepetersen) [inline cpu/cc2538/periph/spi.c]: fixed

**Comment by haukepetersen** [comment]
fixed include issue for the cc2538 i2c driver

**Comment by haukepetersen** [comment]
@lebrush, @kaspar030: what do you say, can we merge this?

**Comment by lebrush** [comment]
Good for me :+1:

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
rebased again. 

@lebrush, @kaspar030 ping

**Comment by lebrush** [comment]
Still looks good to me and still can't merge it :-P

**Comment by haukepetersen** [comment]
@kaspar030: could give a positive review so we can merge?!

**Pull request merged**